### PR TITLE
ci: add UNIX timestamp to version number

### PR DESCRIPTION
**Pull request by ImRodry**
**Please describe the changes this PR makes and why it should be merged:**
Changed the git sha to only include the first 7 characters and added a UNIX timestamp in seconds. This should prevent issues with `npm update` and dependabot being stuck on older versions.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: .github/workflows/publish-dev.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
 
       - name: Publish
         run: |
-          npm version --git-tag-version=false $(jq --raw-output '.version' package.json).$(git rev-parse --verify HEAD)
+          npm version --git-tag-version=false $(jq --raw-output '.version' package.json).$(git rev-parse --short=7 --verify HEAD).$(date +%s)
```
Comment (vaporox) [inline .github/workflows/publish-dev.yml:25]:
```suggestion
          npm version --git-tag-version=false $(jq --raw-output '.version' package.json).$(git rev-parse --short HEAD).$(date +%s)
```
This should be sufficient

Comment (iCrawl) [reply .github/workflows/publish-dev.yml:25]: I'd like to keep `--verify` and the explicit 7 doesn't hurt no one.

Comment (ImRodry) [reply .github/workflows/publish-dev.yml:25]: Crawl asked for SHA 7 and removing =7 gives SHA 8. Also why remove --verify?

Comment (vaporox) [reply .github/workflows/publish-dev.yml:25]: Might be possible that newer git versions have increased the default, in this case that should be fine. Taken from `man git-rev-parse`, `--short` implicitly enables `--verify` so there's no need to add it too.

**Comment by ImRodry** [comment]
Forgot to include a screenshot but this is what it will look like (with the change above)
image

**Review by iCrawl — APPROVED** [review]

**Pull request merged**